src/main/java/com/amazon/dlic/auth/http/jwt/HTTPJwtAuthenticator.java

```diff
@@ -231,30 +248,28 @@ protected String extractSubject(final Claims claims, final RestRequest request)
     @SuppressWarnings("unchecked")
     protected String[] extractRoles(final Claims claims, final RestRequest request) {
     	// no roles key specified
-    	if(rolesKey == null) {
+    	if(rolesKey == null && jsonRolesPath == null) {
     		return new String[0];
     	}
+
+    	// get roles from roles_path if roles_path is specified
+    	if (jsonRolesPath != null){
+    	    try {
+                return RolesUtil.split(JsonPath.using(jsonPathConfig).parse(claims).read(jsonRolesPath));
+            } catch (PathNotFoundException e) {
+                log.error("The provided JSON path {} could not be found ", jsonRolesPath);
+                return new String[0];
+            }
+        }
+
 		// try to get roles from claims, first as Object to avoid having to catch the ExpectedTypeException
     	final Object rolesObject = claims.get(rolesKey, Object.class);
```

Comment:
What will be returned here if instead of setting "roles_path", "roles_key" will be set to the same value?

Reply:
Can you specify more about this comment?

Reply:
Please see my comment regarding deprecating `roles_key`.